### PR TITLE
PIM-4421: call filter to compare values

### PR DESCRIPTION
**Pull request by skeleton**

### CODE REVIEW

File: src/Pim/Component/Connector/Processor/Denormalization/ProductProcessor.php [hunk 1]
```diff
@@ -50,13 +51,20 @@ class ProductProcessor extends AbstractProcessor
...
+    protected $productFilter;
+
     /**
-     * @param StandardArrayConverterInterface       $arrayConverter array converter
-     * @param IdentifiableObjectRepositoryInterface $repository     product repository
-     * @param ProductBuilderInterface               $builder        product builder
-     * @param ObjectUpdaterInterface                $updater        product updater
-     * @param ValidatorInterface                    $validator      product validator
-     * @param ObjectDetacherInterface               $detacher       detacher to remove it from UOW when skip
+     * @param StandardArrayConverterInterface       $arrayConverter     array converter
```
Comment (solivier) [inline src/Pim/Component/Connector/Processor/Denormalization/ProductProcessor.php]: too many spaces

File: src/Pim/Component/Connector/Processor/Denormalization/ProductProcessor.php [hunk 2]
```diff
@@ -345,4 +405,56 @@ protected function getDefaultValues()
...
+
+    /**
+     * This is a temp method to convert "data" keys to "value" keys.
+     * This is an inconsistent between array converters & normalizers. Will be removed with PIM-4246
+     *
+     * @param array $items
+     *
+     * @return array
+     */
+    protected function convertDataToValue(array $items)
```
Comment (solivier) [inline src/Pim/Component/Connector/Processor/Denormalization/ProductProcessor.php]: no 'converters' services for that ?

Comment (skeleton) [inline src/Pim/Component/Connector/Processor/Denormalization/ProductProcessor.php]: That's a temp method, write in a comment:
`This is a temp method to convert "data" keys to "value" keys.
This is an inconsistent between array converters & normalizers. Will be removed with PIM-4246`

File: src/Pim/Component/Connector/Processor/Denormalization/ProductAssociationProcessor.php [hunk 3]
```diff
@@ -61,6 +68,16 @@ public function process($item)
         }
 
         $convertedItem = $this->convertItemData($item);
+
+        if ($this->stepExecution->getJobExecution()->getJobInstance()->getRawConfiguration()['enabledComparison']) {
```
Comment (nidup) [inline src/Pim/Component/Connector/Processor/Denormalization/ProductAssociationProcessor.php]: $this->getConfiguration() ?

File: src/Pim/Component/Connector/Processor/Denormalization/ProductProcessor.php [hunk 4]
```diff
@@ -184,6 +217,26 @@ public function getFamilyColumn()
     }
 
     /**
+     * Set whether or not the comparison between original values and imported values should be activated or not
```
Comment (phaseinducer) [inline src/Pim/Component/Connector/Processor/Denormalization/ProductProcessor.php]: "or not" x2 =)

File: src/Pim/Component/Connector/Processor/Denormalization/ProductProcessor.php [hunk 5]
```diff
@@ -184,6 +217,26 @@ public function getFamilyColumn()
...
+     *
+     * @param bool $enabledComparison
+     */
+    public function setEnabledComparison($enabledComparison)
+    {
+        $this->enabledComparison = $enabledComparison;
+    }
+
+    /**
+     * Whether or not the comparison between original values and imported values should be activated or not
```
Comment (phaseinducer) [inline src/Pim/Component/Connector/Processor/Denormalization/ProductProcessor.php]: "or not" x2
"should be activated" -> " is activated"

**Comment by phaseinducer** [comment]
GTM after behats

File: src/Pim/Component/Connector/Processor/Denormalization/ProductProcessor.php (hunk 2 above)
Comment (solivier) [inline src/Pim/Component/Connector/Processor/Denormalization/ProductProcessor.php]: my bad didn't read the comment :(

**Pull request merged**